### DISCUSSION TITLE
Having trouble getting CLI to work

### DISCUSSION THREAD

**Discussion by jodaalla** [WesSouza/lgtv-ip-control#129] [Q&A]
Having trouble getting the CLI to work and I don't know whether I have missed someting or is just too much of a rookie when it comes to JS libraries or terminals... I'm trying to run the commands from the CLI README but the terminal won't recognize "lgtv-ip-control" as a valid command.
I'm running 64-bit bookworm on a RPI 5 and have installed node, npm, commander, @commander-js/extra-typings (all up to date) and the packages as per the README described.

Anyone got an idea of what I'm missing? Or is it just that the CLI won't work on an RPI with debian?

**Comment by jadoc** [comment]
Are you using the prebuild release, or are you building it yourself?

Currently, the prebuilt release only targets these OS and architecture combinations:
* node18-linux-x64
* node18-macos-x64
* node18-win-x64

As you can see, they are all 64-bit Intel architectures. Your RPI 5 is a 64-bit ARM architecture.

Options:
1) Build the CLI yourself on your own ARM Linux system.
2) Modify the package build workflow to also build for ARM. As the Github build platform is Intel-based, this will require setting up QEMU to emulate ARM. Though, I heard Github may be adding support for ARM actions at some point.
3) Incorporate this tool into Debian. It will then automatically be built for every architecture supported by Debian and be available in the repo. While this would be ideal, it's not a fast process:
https://mentors.debian.net/intro-maintainers/

**Reply by jodaalla** [reply]
I'm using the prebuild. But then I get why it won't work "as is".

But I will try the methods you suggested. Otherwise I can just write a couple of simple JS-script and run them with node. That seemed to work just fine.

Thanks for the help! :)
